api/src/services/files.ts

```diff
@@ -124,6 +125,50 @@ export class FilesService extends ItemsService {
 		return primaryKey;
 	}
 
+	/**
+	 * Sync file from filesystem and update it on database
+	 * If the file doesn't exists on database it will create it
+	 */
+	async sync(relativePath: string): Promise<any> {
+		const res = await this.knex(this.collection).select('*').where({ filename_disk: relativePath }).first();
+
+		if (!res) {
+			const stats = await storage.disk('local').getStat(relativePath);
+			const fileExtension = path.extname(relativePath);
+
+			const payload: Partial<File> & { filename_download: string; storage: string } = {
+				storage: 'local',
+				filename_download: relativePath,
+				type: mime.lookup(relativePath) || 'application/octet-stream',
+				filename_disk: relativePath,
+				title: relativePath.replace(fileExtension, '').replace('-', ' ').toLowerCase(),
+				filesize: stats.size,
+			};
+
+			const primaryKey: PrimaryKey = await this.createOne(payload, { emitEvents: false });
+
+			if (['image/jpeg', 'image/png', 'image/webp', 'image/gif', 'image/tiff'].includes(payload.type || '')) {
+				const buffer = await storage.disk(payload.storage).getBuffer(relativePath);
+				const { height, width, description, title, tags, metadata } = await this.getMetadata(buffer.content);
+
+				payload.height ??= height;
+				payload.width ??= width;
+				payload.description ??= description;
+				payload.title ??= title;
+				payload.tags ??= tags;
+				payload.metadata ??= metadata;
+			}
+
```

Comment:
Wondering if we can make this bit reusable from the regular upload functionality. All the parsing and saving should be the same.